prow/cmd/tackle/ingress.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"github.com/sirupsen/logrus"
+	extensions "k8s.io/api/extensions/v1beta1"
+	networking "k8s.io/api/networking/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/kubernetes"
+)
+
+// https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/20190125-ingress-api-group.md#116
+// NOTE: GCE ingress controller does not target "networking.k8s.io/v1beta1": https://github.com/kubernetes/ingress-gce/issues/770
+const networkingIngressKubeVersion = "v1.16.0"
+
+var runtimeScheme = runtime.NewScheme()
+
+func init() {
+	extensions.AddToScheme(runtimeScheme)
+	networking.AddToScheme(runtimeScheme)
+}
+
+// networkingIngressAvailable checks if the "networking.k8s.io/v1beta1" resource is available based on Kubernetes version.
+func networkingIngressAvailable(kc kubernetes.Interface) bool {
+	serverVersion, err := kc.Discovery().ServerVersion()
```

Comment:
Why not actually check the `ServerResourcesForGroupVersion()` and look for the new type?

Reply:
🤔 I did not realize that the `ServerResourcesForGroupVersion()` method existed. Let me see if using it is a viable option.

Reply:
I have used it before for exactly this - which "Whatever" is the best one to use...

Reply:
Thanks, I will give it a try and update the PR.

Reply:
https://github.com/kubernetes/kubernetes/blob/7a578febe155a7366767abce40d8a16795a96371/pkg/kubectl/cmd/util/generator.go#L212-L228 is a good example use

used like this to let `kubectl run` generate an object the server it is talking to can handle

https://github.com/kubernetes/kubernetes/blob/7a578febe155a7366767abce40d8a16795a96371/pkg/kubectl/cmd/util/generator.go#L153-L156

Reply:
@thockin @liggitt - updated to use `ServerResourcesForGroupVersion()`